core/domain/exp_domain.py

```diff
@@ -1584,11 +1569,45 @@ def create_default_state(
         """
         content_html = (
             feconf.DEFAULT_INIT_STATE_CONTENT_STR if is_initial_state else '')
-        return cls(
-            SubtitledHtml(content_html, {}),
-            [],
-            InteractionInstance.create_default_interaction(
-                default_dest_state_name))
+        interaction = InteractionInstance.create_default_interaction(
+            default_dest_state_name).to_dict()
+
+        interaction = InteractionInstance.from_dict(interaction)
+        return cls(SubtitledHtml(content_html, {}), [], interaction)
+
+    @classmethod
+    def create_default_question_data(cls, default_dest_state_name):
+        """Return a State domain object with default values which can be used
+        for Question.question_data
+
+        Args:
+            default_dest_state_name: str. The default destination state.
+
+        Returns:
+            State. The corresponding State domain object.
+        """
+        interaction = InteractionInstance.create_default_interaction(
+            default_dest_state_name).to_dict()
+        solution = {
+            'answer_is_exclusive': False,
+            'correct_answer': 'helloworld!',
+            'explanation': {
+                'html': 'hello_world is a string',
+                'audio_translations': {}
+            }
+        }
+        interaction['id'] = 'Continue'
```

Comment:
This is interesting, actually. I think it depends on when a question is "created" in the backend.

For explorations, the exploration is created in the backend before the exploration editor is even shown. So the "new" version of the exploration must pass validation, since that is "version 1" of the exploration.

For questions, I think the situation is different. When the creator goes to the question editor page and makes a new question, that question should not be stored in the backend yet. It should just be initialized locally as a frontend dict. Only once the creator tries to save it, should the question be sent to the backend and validated.

This has a couple of implications: the result of "create default question" in the backend can just be passed to the frontend without validation as the "starter template" for the user, and validation can be stricter than it would be otherwise.

What do you think of this approach?

Reply:
Yeah, that's fine. In that case, a create_default_question_data is not needed, right? The existing create_default_question() can be used (which returns empty dict as question_data), and the validations can be done just before actually saving the question (before model.commit). Is that ok?

Reply:
Well, depends. Wouldn't your frontend need a populated question dict?

Reply:
Ya, for that, the existing create_default_state would be fine, as validations are not done anyways.

Reply:
Done. For this, I modified the create_default_question, as mentioned (by calling create_default_state for question_data), and added a separate function in only the test files to create a valid question data, as the validation checks are only done before saving the question in the datastore, the default question framework that a creator sees in the beginning need not be a valid question object.